### PR TITLE
Add nginx project

### PR DESCRIPTION
**Pull request by PaDarochek**

### CODE REVIEW

File: projects/nginx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# Install build dependencies.
+RUN apt-get install -y build-essential libssl-dev
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.30.1/cmake-3.30.1.tar.gz && \
+    tar xf cmake-3.30.1.tar.gz && \
+    cd cmake-3.30.1 && \
+    ./configure && \
+    make
+
+RUN apt-get update && apt-get install -y libpcre3-dev zlib1g-dev mercurial ninja-build liblzma-dev libz-dev binutils libtool
```
Comment (headshog) [inline projects/nginx/Dockerfile:27]: You can combine this command with command on line 20

File: projects/nginx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+FROM sydr/ubuntu20.04-sydr-fuzz
+
+# Install build dependencies.
+RUN apt-get install -y build-essential libssl-dev
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.30.1/cmake-3.30.1.tar.gz && \
```
Comment (headshog) [inline projects/nginx/Dockerfile:21]: Do we need to install this current version of cmake? Apt version doesn't work?

File: projects/nginx/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    tar xf cmake-3.30.1.tar.gz && \
+    cd cmake-3.30.1 && \
+    ./configure && \
+    make
+
+RUN apt-get update && apt-get install -y libpcre3-dev zlib1g-dev mercurial ninja-build liblzma-dev libz-dev binutils libtool
+
+RUN git clone --depth 1 https://github.com/google/libprotobuf-mutator.git
+RUN git clone --depth 1 https://github.com/google/fuzzer-test-suite
+RUN (mkdir LPM && cd LPM && \
```
Comment (headshog) [inline projects/nginx/Dockerfile:31]: Maybe this should be moved to build.sh script?

File: projects/nginx/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        -DLIB_PROTO_MUTATOR_TESTING=OFF -DCMAKE_BUILD_TYPE=Release && ninja)
+
+# Clone target from GitHub. Checkout specified commit. It could be updated later.
+RUN hg clone http://hg.nginx.org/nginx/ -u d1b8568f3042
+
+# Copy build script and additional files for Sydr build.
+COPY build.sh add_sydr.diff make_sydr /
+
+# Build fuzz targets.
+WORKDIR /nginx
```
Comment (headshog) [inline projects/nginx/Dockerfile:42]: Do we need this line? Workdir is changed below

File: projects/nginx/build.sh [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+for fuzzer in objs/*_fuzzer;
+do
+    fuzzer_name=$(basename $fuzzer)
+    fuzzer_name=${fuzzer_name::-7}
+    cp $fuzzer /${fuzzer_name}_cov
+done
+
+# Prepare seed corpora
+cd /
+git clone https://github.com/dvyukov/go-fuzz-corpus.git /go-fuzz-corpus
```
Comment (headshog) [inline projects/nginx/build.sh:138]: You can delete `go-fuzz-corpus` directory after corpora preparation. git clone can be also moved to Dockerfile as it is done in other oss-sydr-fuzz projects.

**Review by headshog — CHANGES_REQUESTED** [review]

File: projects/nginx/http_request.toml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+
```
Comment (headshog) [inline projects/nginx/http_request.toml:1]: Remove this line

File: projects/nginx/Dockerfile (hunk 2 above)
Comment (PaDarochek) [reply projects/nginx/Dockerfile:21]: Apt version doesn't fit for building libprotobuf

File: projects/nginx/Dockerfile (hunk 1 above)
Comment (PaDarochek) [reply projects/nginx/Dockerfile:27]: Done

File: projects/nginx/Dockerfile (hunk 3 above)
Comment (PaDarochek) [reply projects/nginx/Dockerfile:31]: Done

File: projects/nginx/Dockerfile (hunk 4 above)
Comment (PaDarochek) [reply projects/nginx/Dockerfile:42]: Done, deleted

File: projects/nginx/build.sh (hunk 5 above)
Comment (PaDarochek) [reply projects/nginx/build.sh:138]: Done

File: projects/nginx/http_request.toml (hunk 6 above)
Comment (PaDarochek) [reply projects/nginx/http_request.toml:1]: Done

**Review by headshog — APPROVED** [review]

**Review by apach301 — APPROVED** [review]

**Pull request merged**